valohai_yaml/objs/pipelines/execution_node.py

```diff
@@ -1,30 +1,51 @@
+from enum import Enum
 from typing import Any, Dict, List, Optional, Union
 
 from valohai_yaml.lint import LintResult
 from valohai_yaml.objs.pipelines.node import ErrorAction, Node
 from valohai_yaml.objs.pipelines.node_action import NodeAction
-from valohai_yaml.types import LintContext, NodeOverrideDict
+from valohai_yaml.types import LintContext, NodeOverrideDict, SerializedDict
+
+
+class Mode(Enum):
+    """What should happen when error occurs in nodes execution."""
+
+    MERGE = 'merge'  # merge inputs in a node with inputs from step
+    OVERRIDE = 'override'  # default: override inputs in a node
+
+    @classmethod
+    def cast(cls, value: Optional[Union['Mode', str]]) -> 'Mode':
+        if not value:
+            return Mode.OVERRIDE
+        if isinstance(value, Mode):
+            return value
+        value = str(value).lower()
+        return Mode(value)
 
 
 class ExecutionNode(Node):
     """Represents an execution node within a pipeline definition."""
 
     type = 'execution'
+    mode: Mode
 
     def __init__(
-        self,
-        *,
-        name: str,
-        step: str,
-        actions: Optional[List[NodeAction]] = None,
-        override: Optional[NodeOverrideDict] = None,
-        on_error: Union[str, ErrorAction] = ErrorAction.STOP_ALL,
+            self,
+            *,
+            name: str,
+            step: str,
+            actions: Optional[List[NodeAction]] = None,
+            override: Optional[NodeOverrideDict] = None,
+            on_error: Union[str, ErrorAction] = ErrorAction.STOP_ALL,
+
+            mode: Union[str, Mode] = Mode.OVERRIDE
```

Comment:
Please fix your IDE to not do this reindentation :)

Also, add a trailing comma?